server/controllers/finance/debtors/groups/index.js

```diff
@@ -230,24 +230,68 @@ function list(req, res, next) {
 /**
  * GET /debtor_groups/{:uuid}/invoices?balanced=1
  *
- * This function is responsible for getting all invoices of a specified debtor group
- *
  * @function invoices
- * @todo - implement this following the design in debtors/index.js
+ *
+ * @description This function is responsible for getting all invoices of a specified debtor group
+ *
  */
 function invoices(req, res, next) {
   var options = req.query;
 
-  if (options.balanced) { options.balanced = Boolean(options.balanced); }
+  loadInvoices(req.params.uuid)
+  .then(function (rows) {
+    let data = rows;
 
-  res.status(500).send('Unimplemented..');
+    if (options.balanced) {
```

Comment:
Could you make this in `options.balanced` filter using SQL?  Either a `WHERE` or a `HAVING` clause?

It would be much more efficient.